config/database.yml

```diff
@@ -6,7 +6,7 @@ mysql: &MYSQL
   username: root
   password:
   host: localhost
-  port: 3306
+  port: <%= ENV.fetch('DBPORT','3306') %>
```

Comment:
This defaults to returning a string, is that OK?

Reply:
Yep, 3306 is the default port, this avoids us needing to set ENVs in development mode

Reply:
Yeah but doesn't yaml interpret the previous `port: 3306` as an int while `port: <%= ENV.fetch('DBPORT','3306') %>` would return a string? Excuse my yaml ignorance!

Reply:
Ahh, I see what you mean. Actually, in this case, its ruby, or possibly erb knowledge you need.

So `ENV.fetch('DBPORT','3306')` will indeed return a string, either '3306' or the value of DBPORT. However, <%= value %> will interpolate value by calling `.to_s` on it, not `.inspect`, so by the time the yaml parser gets to it, `port: <%= ENV.fetch('DBPORT','3306') %>` will have undergone pre-processing by erubis, and will be `port: 3306`.

That said, I suspect rails probably calls to_i on the value anyway.